File: services/frontend/src/lib/schemas/teams.ts
import z from 'zod';

export const teamResponse = z.object({
  id: z.number(),
  name: z.string(),
  description: z.string().optional().nullable(),
  created_at: z.string(),
  updated_at: z.string(),
});

export type TeamResponse = z.infer<typeof teamResponse>;

export const userTeamsResponse = z.array(teamResponse);

export type UserTeamsResponse = z.infer<typeof userTeamsResponse>;

export const createTeamInput = z.object({
  name: z
    .string()
    .min(1, 'Team name is required')
    .max(255, 'Team name must be less than 255 characters'),
});

export type CreateTeamInput = z.infer<typeof createTeamInput>;

export const createTeamResponse = teamResponse;

export type CreateTeamResponse = z.infer<typeof createTeamResponse>;

File: services/acacia/packages/schemas/teams.go
package schemas

import (
	"acacia/packages/db"
	"errors"

	"github.com/go-playground/validator/v10"
)

type CreateTeamInput struct {
	Name string `json:"name" validate:"required,min=1,max=255"`
}

type TeamResponse struct {
	ID          int64  `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description,omitempty"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

type GetUserTeamsResponse []db.Team

// HandleTeamValidationErrors converts validator errors to user-friendly messages
func HandleTeamValidationErrors(err error) error {
	validationErrors, ok := err.(validator.ValidationErrors)
	if !ok {
		return errors.New("Validation failed")
	}

	for _, e := range validationErrors {
		switch e.Field() {
		case "Name":
			if e.Tag() == "required" {
				return errors.New("Team name is required")
			}
			return errors.New("Team name must be between 1 and 255 characters")
		default:
			return errors.New("Validation failed")
		}
	}

	return errors.New("Validation failed")
}
